theano/gpuarray/extra_ops.py

```diff
@@ -451,24 +451,24 @@ def c_support_code_struct(self, node, nodename):
         return super(GpuCumsum, self).c_support_code_struct(node, nodename) + code
 
 
+@register_opt('fast_compile')
 @op_lifter([CumsumOp])
-def use_gpu_cumsumop(node, ctx_name):
-    if node.inputs[0].dtype == 'float32':
-        axis = node.op.axis
-        x = node.inputs[0]
+@register_opt2([CumsumOp], 'fast_compile')
+def use_gpu_cumsumop(op, ctx_name, inputs, outputs):
+    if inputs[0].dtype == 'float32':
+        axis = op.axis
+        x = inputs[0]
 
         if axis is not None and x.ndim > GpuCumsum.SUPPORTED_NDIMS:
             return None
 
         if axis is None and x.ndim > 1:
             x = x.flatten()
```

Comment:
I think this should call a GPU-only version of `flatten()`, otherwise we are creating nodes back on the CPU.
Please check what the implementation of `flatten` does, and try to build a GPU-only version here, or write a helper function doing that.